javascript/javascript1/week4/get-reply.js

```diff
@@ -0,0 +1,138 @@
+// voice assistant function in javascript
+/* name should be global variable beacause the name is extracted 
+in the first call to getReply and it is displayed in the second call 
+ */
+let name;
+let todo = [];
+function getReply(inputString) {
+  inputString = inputString.trim();
+  if (inputString.search("Hello my name is") == 0) {
+    // this block of code is say hello to user and store user name in the global variable
+    let positionOfName =
+      inputString.search("Hello my name is Benjamin") +
+      "Hello my name is".length;
+    name = inputString.substring(positionOfName);
+    return "Nice to meet you " + name;
+  }
+  if (inputString.search("What is my name") == 0) {
+    // this block of code is dispaly name
+    if (name != null) return "Your name is " + name;
+    else return "Name is not defined";
+  }
+  // the following code is to add activity to "todo list"
+  if (inputString.search("Add") == 0) {
```

Comment:
Another way 
```js
if (inputString.startsWith('add') && inputString.endsWith('to my todo')) {
```
There is also some advanced way using `regular expression` :
```js
/add.+to my todo/i.test(inputString);
```
`i` means `ignore case` to accept capital or lower case
`.+`: `.` means any characters `+` means `one or more`